src/nvim/os/os_defs.h

```diff
@@ -1,10 +1,140 @@
 #ifndef NVIM_OS_OS_DEFS_H
 #define NVIM_OS_OS_DEFS_H
 
+#include <ctype.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
 #ifdef WIN32
 # include "nvim/os/win_defs.h"
 #else
 # include "nvim/os/unix_defs.h"
 #endif
 
+/* The number of arguments to a signal handler is configured here. */
+/* It used to be a long list of almost all systems. Any system that doesn't
+ * have an argument??? */
+#define SIGHASARG
+
+/* List 3 arg systems here. I guess __sgi, please test and correct me. jw. */
+
+#ifdef SIGHASARG
+# ifdef SIGHAS3ARGS
+#  define SIGDEFARG(s)  (int s, int sig2, struct sigcontext *scont)
+#  define SIGDUMMYARG   0, 0, (struct sigcontext *)0
+# else
+#  define SIGDEFARG(s)  (int s)
+#  define SIGDUMMYARG   0
+# endif
+#else
+# define SIGDEFARG(s)  (void)
+# define SIGDUMMYARG
+#endif
```

Comment:
This can probably be simplified a lot... but I'd rather leave that for later, as this is hard enough to review as is :grin: